Added support for several mods

vizthex123 opened the pull request:
- Bigger Reactors (also renamed the file for Extreme Reactors to make it clearer)
- Deeper Caves
- Deep Resonance
- Galosphere (added Salt Blocks since Silver Ore was already done)
- The Outer End
- Project Red
- Regions Unexplored
- RF Tools
- Thermal Series
- XyCraft: World (only Kivi and Aluminum since Xychorium ores don't copy over textures)
- Yung's Cave Biomes

vizthex123 committed:
Added support for several mods

- Bigger Reactors (also renamed the file for Extreme Reactors to make it clearer)
- Deeper Caves
- Deep Resonance
- Galosphere (added Salt Blocks since Silver Ore was already done)
- The Outer End
- Regions Unexplored
- RF Tools
- Thermal Series
- XyCraft: World (only Kivi and Aluminum since Xychorium ores don't copy over textures)

vizthex123 commented:
I'm hoping the mod dev can figure out why Xychorium variants don't have textures (mentioned it #205). I think it uses a custom shader or something that Dynamic Asset Generator isn't properly detecting.

vizthex123 committed:
Fixed Galosphere's salt being a custom type

vizthex123 committed:
Removed custom type for Chalk

vizthex123 committed:
Fixed Slate and Kivi Ores not auto-generating

vizthex123 committed:
Pretty sure this fixes Kivi Ores

I tested it and it seems like it does, but at this point I don't even know anymore.

lukebemish commented:
I'll look into the missing textures; I suspect you're write, they probably have some sort of custom model thingy I can't detect. I'll try to review the rest of this PR when I have a chance, though 1.20.1 is fairly low priority right now so it may be a bit.

vizthex123 commented:
Alright, that sounds good.

Just wanted to toss in what I'd done to save you some time (I figured it probably wasn't too hard to release an update for 1.20.1 if code wasn't changed) and support a couple major-*ish* mods that were missing it.

vizthex123 committed:
Updated the mod support I added

Added `_ore` to each ID and `orename` to a few that were missing it.

vizthex123 committed:
Fixed Aluminum ID and Slate type

Also blacklisted Kivi Aluminum again since I misunderstood how that worked.

I still can't get Slate to blacklist vanilla ores ffs

lukebemish commented:
Sorry I've been so delayed in looking at this! I should have more free time soon so I'll plan on getting it looked over and merged soon-ish.

vizthex123 commented:
noice, thanks.

vizthex123 committed:
Fixed the name for TE's Silver

vizthex123 committed:
Standardized a few ore names I missed

vizthex123 committed:
Finally fixed all the duplicate slate ores

I had an epiphany to just not assign a type to them lol

vizthex123 committed:
Fixed incorrect localized name for lapis

I really should've triple-checked all of this
before forking the repo cuz now i've got a dozen useless commits lol

lukebemish commented on Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/deepercaves.json5:
```diff
@@ -0,0 +1,172 @@
+      block_id: [
+        'cavesanddepths:slate_coal_ore',
+      ],
+      types: [
```
I feel like these ought to have at least one type, to actually generate any variants?

vizthex123 replied:
No, the point is to blacklist it. They're just reskins of vanilla ores, and I didn't want the mod to generate duplicate Slate versions of each one since it's already covered by vanilla ores existing.

I think I told you about it in my issue report?

Either way though, if they're not blacklisted Excavated Variants makes a duplicate version of Slate for all vanilla ores - which is unnecessary & annoying since it already does that for vanilla ores.

lukebemish replied:
What, no. It _won't_ generate duplicate versions if you use the same ore ID, cause the entries from this and the entries from the vanilla coal ore file will get merged. The issue is that you're using the ID `slate_coal_ore` for some reason so it thinks its another ore, cause, well, you _told_ it that it is.

vizthex123 replied:
The problem is that doing so makes it entirely remove the Slate Ores. That's why I set up this blacklist system. It's the only way to get the mod to keep the Slate Ores while not adding duplicates for them.
![image](https://github.com/user-attachments/assets/e971d136-3fcc-4948-b02d-be20d9b22cf1)

vizthex123 replied:
Had issues with it before (that's why I made the workaround), but it seems to be working now. Finally lol.

lukebemish commented on Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/minecraft.json5:
```diff
@@ -160,10 +160,6 @@
   provided_ores: [
     {
       id: 'coal_ore',
```
Uh, what happened to this file?

vizthex123 replied:
I don't even remember lol. I think I re-organized it and accidently copied it over?

lukebemish replied:
This PR shouldn't touch this file if you're not changing something in it

vizthex123 replied:
Yeah, I think I removed the spacing from the multi-line comment or something. It's been months, so I don't remember - but since it's showing no changes, it would've been something minor like that.

lukebemish commented on Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/deepercaves.json5:
```diff
@@ -0,0 +1,172 @@
+  mod_id: 'cavesanddepths',
+  provided_stones: [
+    {
+      id: 'chert',
```
Stone IDs should be prefixed in case two mods add a stone with the same name -- `cavesanddepths_chert`, for instance. The same goes for stuff throughout the PR.

vizthex123 replied:
Alright, I guess I can change it.

lukebemish commented on Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/galosphere.json5:
```diff
@@ -1,11 +1,42 @@
-  provided_stones: [],
+  provided_stones: [
+    {
+      id: 'pastel_pink_salt',
```
Same as above w.r.t stone IDs

lukebemish commented on Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/outer_end.json5:
```diff
@@ -0,0 +1,31 @@
+  mod_id: 'outer_end',
+  provided_stones: [
+    {
+      id: 'ancient_stone',
```
Same as above w.r.t stone IDs

lukebemish commented on Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/regions_unexplored.json5:
```diff
@@ -0,0 +1,31 @@
+  mod_id: 'regions_unexplored',
+  provided_stones: [
+    {
+      id: 'argillite',
```
Same as above w.r.t stone IDs

lukebemish commented on Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/xycraft.json5:
```diff
@@ -0,0 +1,42 @@
+  mod_id: 'xycraft_world',
+  provided_stones: [
+    {
+      id: 'kivi',
```
Same as above w.r.t stone IDs

vizthex123 committed:
Prefixed stone IDs

lukebemish commented on Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/deepresonance.json5:
```diff
@@ -0,0 +1,60 @@
+      ],
+	},
+  ],
+}
```
Can we make sure files have trailing newlines?

vizthex123 replied:
What do you mean? Empty spaces at the end?

I guess I could, but I'm not sure why it should be there (and it seems like other files don't have them).

lukebemish replied:
Eh, you're right, looks like I didn't fix that issue till a newer version, so no need. Just fix the other stuff and it's good

vizthex123 replied:
Fixed them all.

lukebemish commented:
Once the remaining feedback is addressed I'll get this merged. I can handle porting it forward to newer versions myself; that shouldn't be too bad.

vizthex123 committed:
Finally fixed the Slate duplication issue I've been havnig

vizthex123 commented:
Got everything resolved, and yeah I don't mind. If I ever find more mods missing support, I'll open another PR.

If you could check on Xychorium and add it in an update alongside this PR it'd be amazing. Would love to give all XyCraft ores support for this mod.

lukebemish reviewed (changes requested):
Still need to fix whatever bunch of random changes happened to that one file...

vizthex123 commented:
fine, i will re-download it from the github. I still see no changes at all, but whatever.

vizthex123 committed:
Re-downloaded the minecraft.json off of the GitHub

Apparently I re-organized the file to make it more readable when making the new files I added.

lukebemish commented:
Looks good to me; I'll get it merged when I've got some free time

vizthex123 commented:
Alrighty.

I'll add support for Project Red too.

lukebemish commented:
Please do not modify this PR further before I merge it. I'll have to review any modifications you make and that'll set it back. Don't feel bad about splitting stuff into separate PRs

vizthex123 committed:
Added support for Project Red

Can't believe I forgot such a classic mod lol

vizthex123 commented:
Sorry, already did that before you commented.

vizthex123 committed:
Fixed Electrotine not being registered

I'm annoyed I have to make another damn commit for this. Can't believe I forgot to double-check it ;-;

lukebemish commented on Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/projectred.json5:
```diff
@@ -0,0 +1,134 @@
+		en_us: 'Electrotine Ore',
+      },
+      block_id: [
+        'thermal:electrotine_ore',
```
Slightly confused about this -- is there supposed to be a thermal ore in this file?

vizthex123 replied:
No, I did it by accident and fixed it in the next commit (which I forgot to push until now lol). Copied the Thermal file as a base for this one since they share a few ores.

lukebemish reviewed (approved):
Looks good. I'll get it out in the next release -- may be just a bit as I have to get all the stuff for that older version working again and port stuff forwards before the release.

vizthex123 commented:
noice, thanks. Looking forward to it.

vizthex123 committed:
Added support for Yung's Cave Biomes

lukebemish commented:
That last commit was (intentionally) reverted and not included when I merged the PR -- if you want to submit it, it should be a separate PR (in fact compat for different mods should be in general) -- adding more stuff after a PR is approved just means that it has to go through the approval process again. The other changes will be released for 1.20.1 as soon as I've got it ported to newer versions (1.21.x) as well.

vizthex123 commented:
yeah, i couldn't figure out how to open another one when one was already open (i'm really dumb and only use github for backups lol).

but i'll see if i can do it now.

thanks for merging it.
